### PR TITLE
Rewrite Varnish Configuration

### PR DESCRIPTION
**Pull request by JohnFLewis**
Aims to improve readability, and cacheability of requests

### CODE REVIEW

File: modules/varnish/templates/default.vcl [hunk 1]
```diff
@@ -281,99 +154,98 @@ sub recv_purge {
...
-	if (
+# Main MediaWiki Request Handling
+sub mw_request {
+	call rate_limit;
+	call mobile_detection;
+	
+	# Assigning a backend
+	if (req.http.X-Miraheze-Debug ~ "^(mw|test|mwtask)[0-9]{3}\.miraheze\.org$") {
+		# mw101.miraheze.org -> mw101, test101.miraheze.org -> test101, mwtask101.miraheze.org -> mwtask101
+		set req.backend_hint = regsub(req.http.X-Miraheze-Debug, "^((mw|test)[0-9]{3})\.miraheze.org", "\1");
```
Comment (paladox) [inline modules/varnish/templates/default.vcl:165]: Needs to have mwtask here as well.

**Review by paladox — APPROVED** [review]
One comment otherwise looks good to me

File: modules/varnish/templates/default.vcl (hunk 1 above)
Comment (RhinosF1) [reply modules/varnish/templates/default.vcl:165]: It is since 4a70cc7. Are you checking an old commit.

Comment (JohnFLewis) [reply modules/varnish/templates/default.vcl]: Done in 1ad9ac5

**Review by paladox — APPROVED** [review]

**Pull request merged**